controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java

```diff
@@ -50,18 +51,15 @@
      * This method increments the global and Stream-specific counters of Stream creations, initializes other
      * stream-specific metrics and reports the latency of the operation.
      *
-     * @param scope             Scope.
-     * @param streamName        Name of the Stream.
-     * @param minNumSegments    Initial number of segments for the Stream.
-     * @param latency           Latency of the createStream operation.
+     * @param scope                          Scope.
+     * @param streamName                     Name of the Stream.
+     * @param activeSegmentsValueSupplier    Initial number of segments for the Stream.
+     * @param latency                        Latency of the createStream operation.
      */
-    public void createStream(String scope, String streamName, int minNumSegments, Duration latency) {
+    public void createStream(String scope, String streamName, Supplier<Number> activeSegmentsValueSupplier, Duration latency) {
```

Comment:
a supplier that doesnt return a future for activeSegmentCount would mean it is making blocking remote call which will hold up a thread. That is not ideal and detrimental to overall processing

Reply:
Yes agree and as mentioned a separate PR will be focusing on segment count.